website/content/docs/external_plugins/_index.md

```diff
@@ -141,6 +152,36 @@ Below is an example of valid `list` output:
 ]
 ```
 
+If you're able to pre-fetch a method's result as part of the `list` method, then you can include the result as a tuple of `[<method>, <result>]` in the `methods` array. Pre-fetching is a useful way to avoid unnecessary plugin script invocations.
+
+Below is an example that includes pre-fetched method results for a static directory that contains known files and content, but may also support streaming new updates dynamically (by invoking the `stream` method on the script). Note that when `read` content is provided in this manner, the size of that content will be automatically populated in `attributes`.
+
+```json
+[
+  {
+    "name": "mydir",
```

Comment:
I'd maybe add a note that's something like "Notice how `list`'s result matches what would have been returned by `plugin_script list mydir`" so the symmetry's more apparent.